Packages/Core/Editor/Generator/Templates/ConstantTemplateFactory.cs

```diff
@@ -0,0 +1,25 @@
+using System;
+using System.Collections.Generic;
+using UnityEditor;
+
+namespace UnityAtoms.Editor
+{
+    [InitializeOnLoad]
+    public static class ConstantTemplateFactory
```

Comment:
All these factories seem to share the same methods, but they don't implement an interface or force the methods in any other way either. Any solutions for this in C#? Do these need to be static?

Reply:
The way I implemented it, the GetTemplate methods need to be static in order to be added dynamically to the AtomGenerator (in hindsight, perhaps ScriptableObjects carrying Templates would have been better).
Since everything in the Factories was static and I didn't want them to be instantiatable, I opted to make them static.